### ISSUE TITLE
Introduce generic `operation.summary` attribute to represent span name

### ISSUE THREAD

**Issue by trask** [open-telemetry/semantic-conventions#1567]
(and rename `db.operation.name` to `db.operation.command` and replace `db.query.summary` with `operation.name`)

Introducing a generic `operation.name` attribute for span name would allow metrics to stamp this generally useful, low cardinality attribute onto metrics. This would also benefit span to metrics pipelines to have a standard place to put the span name.

It would also provide a way for users to override the span name using something like https://github.com/open-telemetry/oteps/pull/207, which is important for SQL queries which are going to have pretty bad span names unless languages implement complex SQL parsing, and even then, they won't have great span names).

Overriding the span name has also been a highly requested feature for HTTP client spans, where the default span names are also pretty bad.

In database metrics, by replacing `db.query.summary` with the more generic `operation.name`, users would get a consistent database metric drilldown experience across both sql and nosql databases, which is currently a bit lacking because operation and collection name are typically only captured on nosql databases (see #1566).

**Comment by trask** [comment]
discussed in database SIG meeting:

* `operation.name` could make applying https://github.com/open-telemetry/oteps/pull/207 more difficult since it could apply to all nested children, so may need a way to target only database client spans or only "first" nested span
  * there are other ways for users to override `operation.name` such as sql comments
* generic `operation.name` allows us to generically not include it on spans (which would be duplicated with span name) while preserving it on metrics

**Comment by trask** [comment]
Relates to span-to-metrics pipeline: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/spanmetricsconnector/README.md#overview

**Comment by trask** [comment]
Discussed further with @lmolkova.

For database semconv, we are thinking to:

- rename `db.operation.name` to `db.operation.command` (to avoid confusion with having both `operation.name` and `db.operation.name`)
- define `operation.name` using the [existing definition of database span name](https://github.com/open-telemetry/semantic-conventions/blob/v1.30.0/docs/database/database-spans.md#name)
- redefine span name to be `{operation.name}`
- keep `db.query.summary`, but only capture it when different from `operation.name`
  - this allows users to override `operation.name` (and span name), while not losing the query summary

**Comment by trask** [comment]
Discussed in database semconv SIG meeting, and we like `operation.summary` over `operation.name` because:

* We (always?) define the span name as a summary of other attributes
* The word "summarization" has some indication of low-cardinality requirement (at least in my mind...)
* It will be less disruptive to database, messaging, gen_ai semconv where there's already an `*.operation.name` attribute

**Comment by trask** [comment]
We are leaning against introducing this.

Some raw notes:

what does operation.summary get us?
- on spans:
  - duplication issue, db.query.summary is duplicated with span name
    - we could say "don't stamp db.query.summary on spans, only stamp it on metrics"
	  - goes against general (undocumented) convention of metric attributes being a subset of span attributes
	- we could live with duplication in this specific instance
  - is this an issue beyond database client spans (with query parsing)?
- on metrics:
  - consistent drilldown on db.client.operation.duration
    - some dbs populate db.operation.name and db.collection.name, while some dbs populate db.query.summary
    - does this matter if first drilldown is db.system.name?
	- could solve with db.operation.summary?
  - is this an issue beyond db.client.operation.duration metric?
    - others don't have problem with multiple operation names so always have *.operation.name
- used to override the span name? which client span is targeted? could use more obvious "span.name" attribute for this...
  - though it is nice to have this custom name show up somewhere on metrics...
  - downsides to adding this later along with mechanism to perform the override?
- pattern of "*.operation.name" everywhere
  - but it doesn't match span name

**Comment by trask** [comment]
3/14 SIG meeting - decided against this, and to close it out. it's always an option to add in the future if we find compelling use case(s)

**Comment by cheempz** [comment]
Thanks for pointing me to this @trask. While common attributes are now defined for HTTP and Database, it seems maybe contingent on each semconv stabilization effort to enforce this convention? If so, one use case in my mind is it would codify a "transaction" dimension such that traces and metrics for the same operation can be correlated. Exemplars can also link metrics to traces but since they are sampled we won't get the full picture of the metric points that fall under a particular span name / operation.

**Comment by trask** [comment]
hi @cheempz, one of our unwritten rules in semconv is that metric attributes should be a subset of the span attributes (when they are measuring the same operation)

so if you have a specific metric timeseries (set of attributes/values), you can generically search spans for any matching the same set of attributes/values

does this achieve the backend correlation you're looking for?

**Comment by cheempz** [comment]
> hi [@cheempz](https://github.com/cheempz), one of our unwritten rules in semconv is that metric attributes should be a subset of the span attributes (when they are measuring the same operation)
> 
> so if you have a specific metric timeseries (set of attributes/values), you can generically search spans for any matching the same set of attributes/values
> 
> does this achieve the backend correlation you're looking for?

yes it would, thanks very much for the insight (and the issue).
